x-pack/legacy/plugins/apm/server/lib/helpers/apm_ui_indices.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Server } from 'hapi';
+import { merge } from 'lodash';
+import { KibanaConfig } from 'src/legacy/server/kbn_server';
+import { getSavedObjectsClient } from './saved_objects_client';
+
+interface ApmOssIndices {
+  sourcemapIndices: string;
+  errorIndices: string;
+  onboardingIndices: string;
+  spanIndices: string;
+  transactionIndices: string;
+  metricsIndices: string;
+  apmAgentConfigurationIndex: string;
+}
+
+interface ApmUiIndicesSavedObject {
+  [key: string]: any;
+  apm_oss: Partial<ApmOssIndices>;
+}
+
+interface ApmUiIndicesConfig {
+  apm_oss: ApmOssIndices;
+}
+
+const APM_UI_INDICES_SAVED_OBJECT_TYPE = 'apm-ui-indices';
+const APM_UI_INDICES_SAVED_OBJECT_ID = 'apm-ui-indices';
+
+export async function getApmUiIndicesSavedObject(server: Server) {
+  const savedObjectsClient = getSavedObjectsClient(server, 'data');
```

Comment:
I forget: did we settle on whether runtime index configuration should be per user, or across all users? And whether anyone should be able to change it or only admins?